### PR TITLE
Allow tests in root package with Scala Native

### PR DESCRIPTION
**Pull request by lolgab**
Porting of my [PR to Scala Native](https://github.com/scala-native/scala-native/pull/1727)

### CODE REVIEW

**Comment by lolgab** [comment]
I try to give you a little bit of context around this.
Mill (and sbt) runs Scala Native tests by generating a object with a main method that extends `scala.scalanative.testinterface.TestMainBase` from Scala Native test-interface. It should define a two field, a list of Test `Framework`s and and a map from tests to their classes.
If you put your test class or the framework in the empty package, the old code generator would generate:
```scala
package scala.scalanative.testinterface
object TestMain extends TestMainBase {
  override val frameworks = List(new _root_.FrameworkNoPackage, new _root_.pkg.FrameworkInPackage)
  override val tests = Map[String, AnyRef]("ClassName" -> new _root_.ClassName, "pkg.Object" -> _root_.pkg.Object )
  def main(args: Array[String]): Unit =
    testMain(args)
}
```
which is not correct because they are in the _<empty>_ package that can't be referred in JVM languages.
With this PR I put the object in the _<empty>_ package (no package definition) and I add `_root_.` only to classes in packages (when their name has a dot in them). To avoid clashes I renamed `TestMain` to `ScalaNativeTestMain`.
```scala
object ScalaNativeTestMain extends scala.scalanative.testinterface.TestMainBase {
  override val frameworks = List(new FrameworkNoPackage, new _root_.pkg.FrameworkInPackage)
  override val tests = Map[String, AnyRef]("ClassName" -> new ClassName, "pkg.Object" -> _root_.pkg.Object )
  def main(args: Array[String]): Unit =
    testMain(args)
}
```

File: scalanativelib/src/ScalaNativeModule.scala [hunk 1]
```diff
@@ -166,7 +166,8 @@ trait TestScalaNativeModule extends ScalaNativeModule with TestModule { testOute
     val testClassloader =
     new URLClassLoader(runClasspath().map(_.path.toIO.toURI.toURL).toArray,
       this.getClass.getClassLoader)
-    val frameworkInstances = TestRunner.frameworks(testFrameworks())(testClassloader)
+    val 
+     = TestRunner.frameworks(testFrameworks())(testClassloader)
```
Comment (lefou) [inline scalanativelib/src/ScalaNativeModule.scala]: Is a val name missing here?

**Comment by lolgab** [comment]
> Could you please avoid git squashing while we are in review. It makes it unecessary harder to see what I already have seen and what's really new. Git rebase is ok, though. We can squash when we merge the PR.

Sure I can. Sorry for amending the commit!

File: scalanativelib/src/ScalaNativeModule.scala (hunk 1 above)
Comment (lolgab) [reply scalanativelib/src/ScalaNativeModule.scala]: I deleted some text by mistake apparently 🙈

**Comment by lolgab** [comment]
Added another fix when there are no tests it was failing with a `head of empty Seq`.
Added some tests for the bugs fixed by the PR.

**Comment by lolgab** [comment]
This new addition allows me to test fastparse without hacks to avoid modules with no tests in them.

**Comment by LeeTibbert** [comment]
As a ScalaNative user, I say "Thank you for your time"  to  @lolgab" and to  @lihaoyi & the Mill crew. 

I spent some time reading the ScalaNative parts.  Given that ScalaNative is pretty experimental to begin with, the SN parts look good to me.   The SN test framework does tend to fall over dead when
the suites being run have errors, but neither Mill nor this PR should bear the weight of that.

**Comment by LeeTibbert** [comment]
Truth in advertising:  I should put it out there that @lolgab 's work and understanding of SN is way better than mine...

**Comment by lihaoyi** [comment]
@lolgab the failures seems like a spurious/flaky travis thing; could you try pushing a new commit to re-trigger the builds?

**Comment by lefou** [comment]
Merging now. (Hoping someone will come up with a ScalaDoc PR later.)

**Comment by lolgab** [comment]
Thank you very much for merging :)

**Comment by lefou** [comment]
Thanks for your contribution!
